javascript/modules/client/src/client.ts

```diff
@@ -579,50 +593,152 @@ export class Client extends ClientCore implements IClient {
 
   //// PRIVATE METHODS METADATA
 
-  private _getDao(daoAddressOrEns: string): Promise<DaoDetails> {
-    // TODO: Implement actual fetch logic using subgraph.
-
-    if (!daoAddressOrEns) {
-      throw new Error("Invalid DAO address or ENS");
+  private async _getDao(daoAddressOrEns: string): Promise<DaoDetails | null> {
+    if (!isAddress(daoAddressOrEns)) {
+      // TODO
+      // ens resolve
+      throw new Error("Invalid address");
```

Comment:
Can't we easily resolve it here?

Reply:
Yes, but I was getting some errors on testing and didn't want to spend too much time fixing them, I'll enable ens for all the functions once I finish connecting them to subgraph

Reply:
```suggestion
      // TODO: Try resolving an ENS domain
      throw new Error("Invalid address");
```